src/functions.rs

```diff
@@ -2827,6 +2894,64 @@ fn validate_argument<'a>(
+                // If there is no function call or the call is not an asociated one fall through and return
+                // the "standard" error.
+                // If this is the associated call we need to do some more digging to give the user a better
+                // error message.
+                if let Some(func_info) = func_info {
+                    if func_info.is_associated_call {
+                        let mut error = ErrorItem::make_compile_or_internal_error(
+                            &format!(
+                                "Expected type inheriting {} for associated call",
+                                target_argument.param_type.name
+                            ),
+                            arg_typeinfo.get_file().as_ref(),
+                            arg_typeinfo.name.get_range(),
+                            &format!(
+                                "An associated call: '{}' was made for this domain.  That call requires this domain inherit {}",
+                                func_info.name, target_argument.param_type.name
+                            ),
+                        );
+                        // We don't know which class the associated call came from, but that would
+                        // be nice information for the user.  So we search through all the associate class
+                        // annotations, and find the class we are looking for. Then add that to the
+                        // error message.
+                        for annotation in arg_typeinfo.get_annotations() {
+                            if let AnnotationInfo::Associate(association) = annotation {
+                                for resource in &association.resources {
+                                    if func_info
+                                        .class
+                                        .get_name()
+                                        .unwrap_or(&CascadeString::from(""))
+                                        .to_string()
+                                        .ends_with(&(".".to_owned() + resource.as_ref()))
+                                    {
+                                        let associate_type_info = types.get(resource.as_ref());
+                                        // If for some reason we cannot find the type, its file, or its range, just return the error
+                                        // as is and hope the user can figure it out.  Realistically this should never be the case.
+                                        if let Some(associate_type_info) = associate_type_info {
+                                            if let (Some(file), Some(range)) = (
+                                                associate_type_info.get_file(),
+                                                associate_type_info.get_name_range(),
+                                            ) {
+                                                if let ErrorItem::Compile(e) = error {
```

Comment:
Unless you can think of another way, I don't think so unless we also pass in the function map.  We have the name of the function, but none of the file/range info.  I thought printing the name, and pointing to the class is a good middle ground.  But if the call comes from an extension of the class it may be harder to find.

Reply:
Yeah, those seem like pretty reasonable obstacles.  We can leave this as is for now, and maybe someday that can be a future enhancement.